File: src/com/fsck/k9/mail/transport/imap/ImapSettings.java
package com.fsck.k9.mail.transport.imap;

import com.fsck.k9.mail.AuthType;
import com.fsck.k9.mail.ConnectionSecurity;
import com.fsck.k9.mail.store.imap.ImapStore;

/**
 * Settings source for IMAP. Implemented in order to remove coupling between {@link ImapStore} and {@link com.fsck.k9.mail.store.ImapConnection}.
 */
public interface ImapSettings {
    String getHost();

    int getPort();

    ConnectionSecurity getConnectionSecurity();

    AuthType getAuthType();

    String getUsername();

    String getPassword();

    String getClientCertificateAlias();

    boolean useCompression(int type);

    String getPathPrefix();

    void setPathPrefix(String prefix);

    String getPathDelimeter();

    void setPathDelimeter(String delimeter);

    String getCombinedPrefix();

    void setCombinedPrefix(String prefix);
}

File: mail/protocols/imap/src/main/java/com/fsck/k9/mail/store/imap/ImapSettings.kt
package com.fsck.k9.mail.store.imap

import com.fsck.k9.mail.AuthType
import com.fsck.k9.mail.ConnectionSecurity

/**
 * Settings source for IMAP. Implemented in order to remove coupling between [ImapStore] and [ImapConnection].
 */
internal interface ImapSettings {
    val host: String
    val port: Int
    val connectionSecurity: ConnectionSecurity
    val authType: AuthType
    val username: String
    val password: String?
    val clientCertificateAlias: String?
    val useCompression: Boolean
    val clientInfo: ImapClientInfo?

    var pathPrefix: String?
    var pathDelimiter: String?
    fun setCombinedPrefix(prefix: String?)
}
